core/src/main/scala/kafka/server/ReplicaManager.scala

```diff
@@ -278,6 +278,14 @@ class ReplicaManager(val config: KafkaConfig,
     }
   }
 
+  // When ReplicaAlterDirThread finishes replacing a current replica with a future replica, it will
+  // remove the partition from the partition state map. But it will not close itself even if the
+  // partition state map is empty. Thus we need to call shutdownIdleReplicaAlterDirThread() periodically
+  // to shutdown idle ReplicaAlterDirThread
+  def shutdownIdleReplicaAlterLogDirsThread(): Unit = {
```

Comment:
I think it is because the thread class is named `ReplicaAlterLogDirsThread`. I can change it if you still prefer `shutdownIdleAlterReplicaLogDirsThread`.